File: src/ships/calculateDistance.ts
import type { PlanetPosition } from "../data/types/planet.ts";

// Shared Euclidean distance helper, extracted out of calculateTravelTime()
// specifically for the Scanner/Probe amendment's own instruction not to
// duplicate this formula a second time (`performScan()` needs the exact
// same distance math). 2D only -- no z axis, same constraint
// calculateTravelTime() already documented (Phase 5 GDD §2.7).
export function calculateDistance(a: PlanetPosition, b: PlanetPosition): number {
  const dx = b.x - a.x;
  const dy = b.y - a.y;
  return Math.sqrt(dx * dx + dy * dy);
}

File: unity/ProfitableCore/Simulation/CalculateDistance.cs
using Profitable.Core.Schema;

namespace Profitable.Core.Simulation;

// Ports src/ships/calculateDistance.ts. Shared Euclidean distance helper,
// extracted out of CalculateTravelTime so PerformScan reuses the exact
// same distance math -- 2D only, no z axis.
public static class DistanceCalculator
{
    public static double CalculateDistance(PlanetPosition a, PlanetPosition b)
    {
        var dx = b.X - a.X;
        var dy = b.Y - a.Y;
        return Math.Sqrt(dx * dx + dy * dy);
    }
}
